### PR TITLE
Display 'Duplicate VM' Reason for duplicate VMs discovered after the CSV parsing

### PR DESCRIPTION
**Pull request by AparnaKarve**
Display 'Duplicate VM' Reason for duplicate VMs discovered after the CSV parsing.

screen shot 2018-06-07 at 10 34 07 am

screen shot 2018-06-06 at 5 03 21 pm

Also addressed a case where no Reason is being sent by the backend. 
In this situation, 'No information Available' should be displayed instead of a blank text

screen shot 2018-06-08 at 10 16 13 am

Fixes https://github.com/ManageIQ/miq_v2v_ui_plugin/issues/207
https://bugzilla.redhat.com/show_bug.cgi?id=1571960

### CODE REVIEW

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepConstants.js [hunk 1]
```diff
@@ -7,5 +7,7 @@ export const V2V_VM_POST_VALIDATION_REASONS = {
   in_other_plan: __('VM exists in a different migration plan'),
   migrated: __('VM has already been migrated'),
   not_exist: __('VM does not exist'),
-  ok: __('VM available for migration')
+  ok: __('VM available for migration'),
+  duplicate: __('Duplicate VM'),
+  no_info_available: __('No information available')
```
Comment (AllenBW) [inline app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepConstants.js]: In the case of no information, the vm in question is unavailable for migration correct?  If so, we might want to add a few words, to indicate so maaaaybeeeeeeee something similar to `VM unavailable for migration, no information available`

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepReducer.js [hunk 2]
```diff
@@ -17,30 +17,69 @@ const initialState = Immutable({
...
+  manageBlankReason(vm);
+};
+
+const _formatValidVms = vms => {
+  const uniqueIds = vms && [...new Set(vms.map(value => value.id))];
+  let vIndex = 0;
+  return (
+    vms &&
+    vms.map(v => {
+      vIndex += 1;
```
Comment (AllenBW) [inline app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepReducer.js]: no need to keep a counter here if on line 49 yah do something like `vms.map((v, index)  => {`

then on line 54 `index + 1` will give you that plus one offset

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepReducer.js [hunk 3]
```diff
@@ -17,30 +17,69 @@ const initialState = Immutable({
...
+  }
+};
+
+const manageBlankReason = vm => {
+  if (!vm.reason) {
+    vm.reason = V2V_VM_POST_VALIDATION_REASONS.no_info_available;
+  }
+};
+
+const manageOddCSVImportErrors = (vm, vmIndex, uniqueIds) => {
```
Comment (AllenBW) [inline app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepReducer.js]: I love this function name ❤️ 🌮 😋

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepReducer.js (hunk 2 above)
Comment (AparnaKarve) [reply app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepReducer.js]: Thanks! That's a good tip :)

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepConstants.js (hunk 1 above)
Comment (AparnaKarve) [reply app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepConstants.js]: Done in 9774ad0

File: app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepReducer.js (hunk 2 above)
Comment (AparnaKarve) [reply app/javascript/react/screens/App/Overview/screens/PlanWizard/components/PlanWizardVMStep/PlanWizardVMStepReducer.js]: 9774ad0

**Pull request merged**